modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/OsgiAssetManager.java

```diff
@@ -197,6 +238,22 @@ public void setAssetStore(AssetStore assetStore) {
     this.assetStore = assetStore;
   }
 
+  public synchronized void addRemoteAssetStore(RemoteAssetStore assetStore) {
+    if (null == delegate) {
+      remotes.add(assetStore);
+    } else {
+      delegate.addRemoteAssetStore(assetStore);
+    }
+  }
+
+  public synchronized void removeRemoteAssetStore(RemoteAssetStore assetStore) {
+    if (null != delegate) {
+      delegate.removeRemoteAssetStore(assetStore);
+    } else {
+      logger.warn("Unable to remove remote store of type {} because delegate is null!", assetStore.getStoreType());
```

Comment:
You could have the asset store removed before activate() was called. Very unlikely, but possible :)

Reply:
I feel like that's so unlikely that we should leave this here.  If it becomes an issue then we need to explore how to properly log this :)